src/ApiStatus.ts

```diff
@@ -100,15 +108,20 @@ export default class ApiStatus {
     }
 
     private async getApiStatus(): Promise<StatusEmbedState> {
-        // cache embed state
-        const timeDiff = Date.now() - this.lastCheckTime;
-        if (timeDiff > this.sharedSettings.apiStatus.checkInterval) {
-            const apiStatus = await this.apiStatusAPI.getApiStatus();
-            this.currentStatus = this.parseApiStatus(apiStatus);
-            this.lastCheckTime = Date.now();
-        }
+        try {
+            // cache embed state
+            const timeDiff = Date.now() - this.lastCheckTime;
+            if (timeDiff > this.sharedSettings.apiStatus.checkInterval) {
+                const apiStatus = await this.apiStatusAPI.getApiStatus();
+                this.currentStatus = this.parseApiStatus(apiStatus);
+                this.lastCheckTime = Date.now();
+            }
+            return this.currentStatus;
 
-        return this.currentStatus;
+        } catch (error) {
+            console.error(`Error occurred while making a request to the apistatus api: ${error}`);
+            return this.currentStatus;

```

Comment:
Due to the catch here this method will never reject. Which means that the handlers above will never reach the error case ;)

Reply:
Also, bit unrelated, but it looks like both this class and the api status API cache the result. Maybe we can remove the caching here?